src/Microsoft.AspNetCore.SignalR.Client/HttpConnectionOptions.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Net.WebSockets;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.AspNetCore.Http.Connections;
+
+namespace Microsoft.AspNetCore.SignalR.Client
+{
+    public class HttpConnectionOptions
+    {
+        internal X509CertificateCollection _clientCertificates;
+        internal IDictionary<string, string> _headers;
+        internal CookieContainer _cookies;
+
+        public Uri Url { get; set; }
+        public TransportType? Transport { get; set; }
+        public Func<HttpMessageHandler, HttpMessageHandler> MessageHandlerFactory { get; set; }
+        public bool? UseDefaultCredentials { get; set; }
+        public ICredentials Credentials { get; set; }
+        public IWebProxy Proxy { get; set; }
+        public Func<string> AccessTokenFactory { get; set; }
+        public Action<ClientWebSocketOptions> WebSocketOptions { get; set; }
+
+        public void AddClientCertificate(X509Certificate clientCertificate)
```

Comment:
I'm not a fan of these methods. They make it harder to do general purpose collection copying (which can be useful for proxying things)